scripts/install-nix-from-closure.sh

```diff
@@ -1,34 +1,346 @@
+}
+
+errorRevert() {
+    message="$1"
+    color="$red"
+    prefix='Error'
+    >&2 print
+}
+
+contactme() {
+    print '
+
+    If you hit a bug, we would love to help.
+
+    You can, search/open an issue at https://github.com/nixos/nix/issues
+
+    Feel free to contact the team,
+     - on IRC: #nixos on irc.freenode.net
+     - on twitter: @nixos_org
+     - on Matrix: https://riot.im/app/#/room/#nix:matrix.org
+'
+}
+
+}
+###############################
+###  Program stage functions
+###############################
+{
+checkingRequirements() {    # This function must have checks only. Without making ANY changes to the system.
+    if ! [ -e "$self/.reginfo" ]; then
+        error "
+
+    Installer is incomplete ($self/.reginfo is missing)
+    "
+    fi
 
-set -e
+    if [ "$(id -u)" -eq 0 ]; then
+        # TODO: Reword after scripts integration and option switching becomes clear.
+        notice "
 
-dest="/nix"
-self="$(dirname "$0")"
-nix="@nix@"
-cacert="@cacert@"
+    Install executed for ROOT.
 
+    Doing classic Linux package manager mode.
+    In Nix this mode is called: single-user mode for root.
 
-if ! [ -e "$self/.reginfo" ]; then
-    echo "$0: incomplete installer (.reginfo is missing)" >&2
-fi
+    Nix can do multi-user mode, and manage package trees for users independently.
+    "
+    fi
 
-if [ -z "$USER" ]; then
-    echo "$0: \$USER is not set" >&2
-    exit 1
-fi
+    # In case if USER not set
+    if [ -z "$USER" ]; then
+        notice "
 
-if [ -z "$HOME" ]; then
-    echo "$0: \$HOME is not set" >&2
-    exit 1
-fi
+    Environment variable USER is not set.
+    Maybe running inside container.
+    "
+        USER="$(id -u -n)"    # id is POSIX
+        print "Detected username: $USER"
+    fi
 
-# macOS support for 10.10 or higher
-if [ "$(uname -s)" = "Darwin" ]; then
-    if [ $(($(sw_vers -productVersion | cut -d '.' -f 2))) -lt 10 ]; then
-        echo "$0: macOS $(sw_vers -productVersion) is not supported, upgrade to 10.10 or higher"
-        exit 1
+    if [ -z "$HOME" ]; then
+        error "
+
+    Environment variable HOME is not set.
+    "
+    fi
+
+    # macOS support for 10.10 or higher
+    if [ "$(uname -s)" = 'Darwin' ]; then
+        if [[ $(($(sw_vers -productVersion | cut -d '.' -f 2))) -lt 10 ]]; then
+            error "macOS $(sw_vers -productVersion) is not supported, upgrade to 10.10 or higher"
+        fi
+        notice 'Switching to the Multi-User Darwin Installer'
+        exec "$self/install-darwin-multi-user"
+        exit 0
+    fi
+
+    if [ ! -e "$HOME" ]; then
+        error "
+
+    Home directory $HOME does not exist.
+    "
+    fi
+
+    if [ ! -d "$HOME" ]; then
+        error "
+
+    Home directory $HOME is not a directory.
+    "
+    fi
+
+    if [ ! -w "$HOME" ]; then
+        error "
+
+    Home directory $HOME is not writable for user $USER. No deal.
+    "
+    fi
+
+    if [ "$(ls -ld "$HOME" | cut -d' ' -f 3)" != "$USER" ]; then    # is POSIX, no `test -O`
```

Comment:
On Alpine/BusyBox the cut operation needs `-f6` because of multiple spaces in between fields. Perhaps use:
```
ls -ld "$HOME" | awk '{print $3}'
```

Reply:
Thank you.

Yes. You are right.
I also was also thinking about changing to `awk`, and here you send with a proof.